src/lib.rs

```diff
@@ -158,16 +158,22 @@ impl HcSr04 {
     /// Perform **distance measurement**.
     ///
     /// Returns `Ok` variant if measurement succedes. Inner `Option` value is `None` if no object
-    /// is present within maximum measuring range (*4m*); otherwhise, on `Some` variant instead,
+    /// is present within maximum measuring range (*4m*); otherwise, on `Some` variant instead,
     /// contained value represents distance expressed as the specified `unit`
     /// (**unit of measure**).
     pub fn measure_distance(&mut self, unit: Unit) -> Result<Option<f32>> {
         self.trig.set_high();
         thread::sleep(Duration::from_micros(10));
         self.trig.set_low();
-
-        // Wait for the `RisingEdge` by ensuring the resulting level is `Level::High`.
-        while self.echo.poll_interrupt(false, None)? != Some(Level::High) {}
+        
+        loop {
+            match self.echo.poll_interrupt(false, Some(self.timeout * 3)) {
```

Comment:
Use smaller timeout?
```suggestion
            match self.echo.poll_interrupt(false, Some(self.timeout * 1.1)) {
```